### PR TITLE
React UI: Batch of fixes

### PR DESCRIPTION
**Pull request by bsekachev**
- Added validations for input number in different places
- Fixed some UI crashes
- Fixed minor bug with merging
- Added ability to undo points when editing
- Removed some old code

### CODE REVIEW

File: cvat-canvas/src/typescript/editHandler.ts [hunk 1]
```diff
@@ -99,6 +99,12 @@ export class EditHandlerImpl implements EditHandler {
             if (e.which === 1) {
                 mouseX = e.clientX;
                 mouseY = e.clientY;
+            } else if (e.which === 3 && this.editLine) {
+                if (this.editData.state.shapeType === 'points'
+                    || this.editLine.attr('points').split(' ').length > 2
+                ) {
+                    (this.editLine as any).draw('undo');
+                }
```
Comment (ActiveChooN) [inline cvat-canvas/src/typescript/editHandler.ts]: Referenced to developer.mozilla.org/ru/docs/Web/API/MouseEvent/which, MouseEvent.which is deprecated, so it maybe better to use MouseEvent.button

File: cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover.tsx [hunk 2]
```diff
@@ -117,7 +113,15 @@ function DrawShapePopoverComponent(props: Props): JSX.Element {
                         </Col>
                         <Col span={10}>
                             <InputNumber
-                                onChange={onChangePoints}
+                                onChange={(value: number | undefined) => {
+                                    if (typeof (value) === 'number') {
+                                        onChangePoints(Math.floor(
+                                            clamp(value, minimumPoints, Number.MAX_SAFE_INTEGER),
```
Comment (ActiveChooN) [inline cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover.tsx]: BTW, why is there two implementation of the clamp function? One in `utils/math.ts` other in `cvat-core/src/api.js`.

Comment (bsekachev) [reply cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/draw-shape-popover.tsx]: Despite everything ``core`` and ``ui`` are independent modules and can be divided in different repositories in the future

File: cvat-canvas/src/typescript/editHandler.ts (hunk 1 above)
Comment (bsekachev) [reply cvat-canvas/src/typescript/editHandler.ts]: I didn't know. If it is, let's use button instead

**Pull request merged**